Decode/EventHeaderEnumerator.cs

```diff
@@ -1092,19 +1092,23 @@ public EventHeaderEventInfo GetEventInfo(ReadOnlySpan<byte> eventDataSpan)
         /// </para>
         /// </summary>
         /// <exception cref="InvalidOperationException">Called in invalid State.</exception>
-        public PerfItemType GetItemType()
+        public PerfItemMetadata GetItemMetadata()
         {
             if (m_state <= EventHeaderEnumeratorState.BeforeFirstItem)
             {
                 throw new InvalidOperationException(); // PRECONDITION
             }
 
-            return new PerfItemType(
+            var isScalar =
+                m_state < EventHeaderEnumeratorState.ArrayBegin ||
+                m_state > EventHeaderEnumeratorState.ArrayEnd;
+            return new PerfItemMetadata(
                 m_byteReader,
                 m_fieldType.Encoding,
                 m_fieldType.Format,
+                isScalar,
                 m_elementSize,
-                m_state == EventHeaderEnumeratorState.Value ? (ushort)1 : m_stackTop.ArrayCount,
+                isScalar ? (ushort)1 : m_stackTop.ArrayCount,
                 m_fieldType.Tag);
```

Comment:
Note: We need to force ElementCount to 1 for scalars. Previously we were only considering Value state as scalar, but there are other scalar states (StructBegin, StructEnd) that should have also been getting ElementCount set to 1. Fix with an accurate determination of IsScalar.